unitstatushandler/unitstatushandler.go

```diff
@@ -175,6 +175,12 @@ func New(
 		systemQuotaAlertChannel: systemQuotaAlertProvider.GetSystemQuoteAlertChannel(),
 	}
 
+	// Initialize containers in order to avoid providing nils to cloud
+	instance.unitStatus.Components = make([]cloudprotocol.ComponentStatus, 0)
```

Comment:
Add all these into `resetUnitStatus` function and call this function here.

Reply:
done